src/measure_theory/function/strongly_measurable.lean

```diff
@@ -240,36 +397,408 @@ protected lemma inf [has_inf β] [has_continuous_inf β] (hf : strongly_measurab
   strongly_measurable f ↔ measurable f :=
 ⟨λ h, h.measurable, λ h, measurable.strongly_measurable h⟩
 
+lemma _root_.strongly_measurable_id [topological_space α] [metrizable_space α]
+  [opens_measurable_space α] [second_countable_topology α] :
+  strongly_measurable (id : α → α) :=
+measurable_id.strongly_measurable
+
 end second_countable_strongly_measurable
 
+/-- A function is strongly measurable if and only if it is measurable and has separable
+range. -/
+theorem _root_.strongly_measurable_iff_measurable_separable {m : measurable_space α}
+  [topological_space β] [metrizable_space β] [measurable_space β] [borel_space β] :
+  strongly_measurable f ↔ (measurable f ∧ is_separable (range f)) :=
+begin
+  refine ⟨λ H, ⟨H.measurable, H.is_separable_range⟩, _⟩,
+  rintros ⟨H, H'⟩,
+  letI := metrizable_space_metric β,
+  let g := cod_restrict f (closure (range f)) (λ x, subset_closure (mem_range_self x)),
+  have fg : f = (coe : closure (range f) → β) ∘ g, by { ext x, refl },
+  have T : measurable_embedding (coe : closure (range f) → β),
+  { apply closed_embedding.measurable_embedding,
+    exact closed_embedding_subtype_coe is_closed_closure },
+  have g_meas : measurable g,
+  { rw fg at H, exact T.measurable_comp_iff.1 H },
+  haveI : second_countable_topology (closure (range f)),
+  { suffices : separable_space (closure (range f)),
+      by exactI uniform_space.second_countable_of_separable _,
+    exact (is_separable.closure H').separable_space },
+  have g_smeas : strongly_measurable g := measurable.strongly_measurable g_meas,
+  rw fg,
+  exact continuous_subtype_coe.comp_strongly_measurable g_smeas,
+end
+
+/-- A continuous function is strongly measurable when either the source space or the target space
+is second-countable. -/
+lemma _root_.continuous.strongly_measurable [measurable_space α]
+  [topological_space α] [opens_measurable_space α]
+  {β : Type*} [topological_space β] [metrizable_space β] [h : second_countable_topology_either α β]
+  {f : α → β} (hf : continuous f) :
+  strongly_measurable f :=
+begin
+  letI : measurable_space β := borel β,
+  haveI : borel_space β := ⟨rfl⟩,
+  casesI h.out,
+  { rw strongly_measurable_iff_measurable_separable,
+    refine ⟨hf.measurable, _⟩,
+    rw ← image_univ,
+    exact (is_separable_of_separable_space univ).image hf },
+  { exact hf.measurable.strongly_measurable }
+end
+
+/-- If `g` is a topological embedding, then `f` is strongly measurable iff `g ∘ f` is. -/
+lemma _root_.embedding.comp_strongly_measurable_iff {m : measurable_space α}
+  [topological_space β] [metrizable_space β] [topological_space γ] [metrizable_space γ]
+  {g : β → γ} {f : α → β} (hg : embedding g) :
+  strongly_measurable (λ x, g (f x)) ↔ strongly_measurable f :=
+begin
+  letI := metrizable_space_metric γ,
+  letI : measurable_space β := borel β,
+  haveI : borel_space β := ⟨rfl⟩,
+  letI : measurable_space γ := borel γ,
+  haveI : borel_space γ := ⟨rfl⟩,
```

Comment:
Nothing to change here, but now I understand the need for that `borelize` tactic coming in another PR :)